ra2ce/analysis/damages/damage_calculation/damage_network_return_periods.py

```diff
@@ -93,98 +96,107 @@ def main(self, damage_function: DamageCurveEnum, manual_damage_functions):
+            return_periods=return_periods
+        )
 
-        if mode == RiskCalculationModeEnum.DEFAULT:
-
-            _to_integrate = self.rework_damage_data_default(_to_integrate)
-            _risk = self.integrate_df_trapezoidal(_to_integrate.copy())
-            self.gdf["risk"] = _risk
-
-        elif mode == RiskCalculationModeEnum.CUT_FROM_YEAR:
-            """
-            In this mode, the integration mimics the presence of a flood protection
-            """
-            _rps = list(_to_integrate.columns)
-
-            if year <= min(_rps):
-                raise ValueError(
-                    """
-                RA2CE cannot calculate risk in 'cut_from' mode if 
-                Return period of the cutoff ({}) <= smallest available return period ({})
-                Use 'default' mode or 'triangle_to_null_mode' instead.
-                                    """.format(
-                        year, min(_rps)
-                    )
-                )
+        for vulnerability_curve_name, _to_integrate in _to_integrate_dict.items():
+            if mode == RiskCalculationModeEnum.DEFAULT:
 
-            elif (
-                min(_rps) < year < max(_rps)
-            ):  # if protection level is between min and max RP
-                _to_integrate = self.rework_damage_data_cut_from(_to_integrate, year)
+                _to_integrate = self.rework_damage_data_default(_to_integrate)
                 _risk = self.integrate_df_trapezoidal(_to_integrate.copy())
 
-            elif year >= max(
-                _rps
-            ):  # cutoff is larger or equal than largest return period
-                # risk is return frequency of cutoff
-                # times the damage of the most extreme event
-                _to_integrate = _to_integrate.fillna(0)
-                _risk = _to_integrate[max(_to_integrate.columns)] / year
-
-            self.gdf["risk"] = _risk
-
-        elif mode == RiskCalculationModeEnum.TRIANGLE_TO_NULL_YEAR:
-            """
-            In this mode, an extra data point with zero damage is added at some distance from the smallest known RP,
-            and the area of the Triangle this creates is also calculated
-            """
-
-            _rps = list(_to_integrate.columns)
-
-            if year >= min(_rps):
-                raise ValueError(
-                    """
-                RA2CE cannot calculate risk in 'triangle_to_null' mode if 
-                Return period of the triangle ({}) >= smallest available return period ({})
-                Use 'default' mode or 'cut_from' instead.
-                                    """.format(
-                        year, min(_rps)
+            elif mode == RiskCalculationModeEnum.CUT_FROM_YEAR:
+                """
+                In this mode, the integration mimics the presence of a flood protection
+                """
+                _rps = list(_to_integrate.columns)
```

Comment:
As someone who is looking at damages module for the first time, it is quite hard to follow and understand what is going on. Therefore reviewing is also complicated.

maybe some variables should be explicitly named or add comment to explain what they are:
- what is _rps ?
- what is "RP" ?

Reply:
_rps: return periods
RP: Return Period